docs/upgrading.md

```diff
@@ -72,6 +72,32 @@ If you're using it because it's "fast", please look into adoption the non-standa
 +);
 ```
 
+### `Builder` API is now `@immutable`
+
+`\Lcobucci\JWT\Builder` interface alongside with its default implementation `\Lcobucci\JWT\Token\Builder` are now marked `@immutable`.
+
+If you are using it for example with the `JwtFacade` ensure now to use the returned new `Builder` instance:
+
+```diff
+ $token = (new JwtFacade())->issue(
+     new Sha256(),
+     $key,
+     static function (
+         Builder $builder,
+         DateTimeImmutable $now
+     ): Builder {
+-        $builder->issuedBy('https://api.my-awesome-app.io');
+-        $builder->permittedFor('https://client-app.io');
+-        $builder->expiresAt($issuedAt->modify('+10 minutes'));
++        $builder = $builder->issuedBy('https://api.my-awesome-app.io');
++        $builder = $builder->permittedFor('https://client-app.io');
++        $builder = $builder->expiresAt($now->modify('+10 minutes'));
```

Comment:
I wouldn't format my code like this, I made it so here to force the user read the diff and think about it.

Reply:
Maybe we can add a note about that on the upgrading guide?

Reply:
About what exactly?

Reply:
That the format only illustrates the immutability factor and we don't expect people to code like that